sw/device/silicon_creator/imm_rom_ext/imm_rom_ext.c

```diff
@@ -4,7 +4,59 @@
 
 #include "sw/device/silicon_creator/imm_rom_ext/imm_rom_ext.h"
 
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/base/boot_measurements.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/cert/dice_chain.h"
+#include "sw/device/silicon_creator/lib/dbg_print.h"
+#include "sw/device/silicon_creator/lib/drivers/pinmux.h"
+#include "sw/device/silicon_creator/lib/drivers/rnd.h"
+#include "sw/device/silicon_creator/lib/drivers/uart.h"
+#include "sw/device/silicon_creator/lib/epmp_state.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/manifest.h"
+#include "sw/device/silicon_creator/lib/shutdown.h"
+#include "sw/device/silicon_creator/rom_ext/rom_ext_manifest.h"
+
+OT_WARN_UNUSED_RESULT
+static rom_error_t imm_rom_ext_start(void) {
+  // Check the ePMP state.
+  HARDENED_RETURN_IF_ERROR(epmp_state_check());
+  // Check sec_mmio expectations.
+  // We don't check the counters since we don't want to tie ROM_EXT to a
+  // specific ROM version.
+  sec_mmio_check_values(rnd_uint32());
+
+  // Initialize Immutable ROM EXT.
+  sec_mmio_next_stage_init();
+  // Configure UART0 as stdout.
+  pinmux_init_uart0_tx();
+  uart_init(kUartNCOValue);
+
+  dbg_printf("IMM_ROM_EXT v0.1\r\n");
+
+  // Establish our identity.
+  const manifest_t *rom_ext = rom_ext_manifest();
+  HARDENED_RETURN_IF_ERROR(dice_chain_init());
+  // TODO: Move UDS cert check to mutable ROM_EXT.
+  HARDENED_RETURN_IF_ERROR(dice_chain_attestation_silicon());
+  HARDENED_RETURN_IF_ERROR(
```

Comment:
In the ROM_EXT, we have to initialize ownership after the silicon stage but before the creator stage.  That is because we bind the ownership configuration to the chip with a keymgr derived KMAC key.

I think at a minimum, we need to generate a sealing key for KMAC here.  See `ownership_seal_init` on the `earlgrey_1.0.0` branch.

Reply:
Added `ownership_seal_init` call between the silicon and the creator stages.